llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -2159,6 +2167,30 @@ bool AArch64LoadStoreOpt::tryToPairLdStInst(MachineBasicBlock::iterator &MBBI) {
     // Keeping the iterator straight is a pain, so we let the merge routine tell
     // us what the next instruction is after it's done mucking about.
     auto Prev = std::prev(MBBI);
+
+    // Fetch the memoperand of the load/store that is a candidate for
+    // combination.
+    MachineMemOperand *MemOp = MI.memoperands().front();
```

Comment:
I think front() might assert if the array is empty. Can this use `MI.memoperands_empty() ? nullptr : MI.memoperands().front()`?

Reply:
Sure, pushing real soon.